### PR TITLE
txconfirm: fix zero-output CPFP child when fee-input change is sub-dust

### LINKED ISSUE #580: txconfirm: CPFP child can have zero outputs when fee-input change falls below dust → FinalizePsbt rejects, unroll wedges
## Summary

`BuildCPFPChild` can produce a CPFP child transaction with **zero outputs**, which
LND's `FinalizePsbt` then rejects with `PSBT packet must contain at least one
output`. When this happens during an unroll fee-bump, the unroll can never
confirm: the package child can't be signed, and the parent unroll-node tx pays
sub-relay-fee on its own so a direct broadcast also bounces with `min relay fee
not met`.

Observed in production on signet while an abandoned vHTLC
(`refund_without_receiver`, originally issue #538) was escalating to an on-chain
unilateral refund. The wallet had ~2 BTC confirmed, so this is **not** a balance
problem — it's a fee-input selection / child-builder mismatch.

## Root cause

`txconfirm/broadcaster.go`:

- `selectFeeInput` (line 949) / `SelectFeeInput` (line 1427) select the
  **smallest** confirmed UTXO whose value is `>= totalFee` (line 979:
  `if utxo.Amount < minAmount { continue }`, with `minAmount == totalFee`).
- `BuildCPFPChild` (line 1401) computes
  `changeValue = feeInput.Output.Value - totalFee` and only adds the change
  output when `changeValue >= DustLimit` (330 sat) (line 1407).

The CPFP child has two inputs (the 0-value/ephemeral anchor + the fee input) and
exactly **one** potential output: the change. So when the selected fee input's
value falls in the window `[totalFee, totalFee + DustLimit)`, `changeValue` is in
`[0, 330)`, the change output is dropped, and the child ends up with **no
outputs at all**.

This is not a rare corner: `selectFeeInput` deliberately picks the *smallest*
qualifying UTXO (line 983: `if best == nil || utxo.Amount < best.Amount`), which
is precisely the input most likely to be just barely above `totalFee` and thus
land in the sub-dust change window.

## Symptom / logs

```
TXCF: trackedtx(<unroll-node>) AwaitingConfirmation -> FeeBumping
TXCF: CPFP unavailable; broadcasting parent directly
  err="finalize PSBT: LND FinalizePsbt: rpc error: code = Unknown desc =
       error finalizing PSBT: PSBT packet must contain at least one output"
  stage=sign_cpfp_child label=unroll-node-<txid>
TXCF: Fee bump failed, will retry
  err="cpfp fee input unavailable: sign_cpfp_child: ... PSBT packet must contain
       at least one output; direct parent broadcast: broadcast <txid>: ...
       transaction rejected by the mempool because of low fees: min relay fee
       not met"
```

Example stuck unroll-node txids on signet:
- `14d3fb1b78d00c0bf86471f467e2c7e28d85d2f5f1b4f83a77cc392e5f3bf8e5`
- `7e631a641b71315266bb033859bb72e4accb54b31698c479d283c8a7a9176c7d`

## Suggested fix

Two layers (defense in depth):

1. **Selection threshold** — require the fee input to cover the fee *plus* a
   spendable change output: pass `minValue = totalFee + DustLimit` to
   `selectFeeInput` / `SelectFeeInput`, so the resulting `changeValue` is always
   `>= DustLimit` and a change output is always emitted.

2. **Builder guard** — make `BuildCPFPChild` return an explicit sentinel error
   (e.g. `ErrFeeInputProducesDust`) when `changeValue < DustLimit`, instead of
   silently returning a zero-output tx. The caller can then exclude that outpoint
   and re-select a larger input rather than handing an invalid PSBT to
   `FinalizePsbt`.

Layer 1 prevents the bad selection; layer 2 ensures we never again silently
construct a degenerate child if some other path computes the fee differently.

## Impact

Blocks on-chain unroll / unilateral exit fee-bumping whenever the smallest
qualifying fee UTXO lands in the dust window. On signet this wedged the
abandoned #538 vHTLC's `refund_without_receiver` unroll indefinitely.

### PR DESCRIPTION
**Pull request by Roasbeef**
## Summary

Fixes #580.

`selectFeeInput` picks the **smallest** confirmed UTXO with `value >= totalFee`,
but `BuildCPFPChild` only emits a change output when `value - totalFee >=
DustLimit` (330). A fee input whose value lands in the window `[totalFee,
totalFee + DustLimit)` therefore passes selection yet yields a CPFP child with
**no outputs** — its only other input is the parent's zero-value ephemeral
anchor — which LND's `FinalizePsbt` rejects (`PSBT packet must contain at least
one output`). The 0-fee TRUC parent then can't relay on its own (`min relay fee
not met`), so on-chain unrolls wedge indefinitely.

Observed on signet: the wallet had ~97M sat across 18 UTXOs, but was littered
with near-dust outputs (`357 / 798 / 944`, plus five `294`s). For a few-hundred-sat
unroll-node fee, the selector grabbed one of the tiny ones, leaving sub-dust
change → zero-output child. The large UTXOs (78M / 12M / 3M) were never reached
because a tiny one already "qualified".

## Fix (two layers)

1. **Selection threshold** — select for `totalFee + DustLimit` so the chosen fee
   input always leaves a spendable change output. Near-dust UTXOs are skipped and
   selection lands on one that yields valid change.
2. **Builder guard** — `BuildCPFPChild` now returns `ErrFeeInputProducesDust`
   when the remainder is below dust, instead of silently constructing a
   zero-output transaction. Defense-in-depth against a post-selection fee top-up
   (mixed-type wallets) eroding the dust margin.

## Tests

- Updated `TestBroadcasterHelperFunctions` "build cpfp child": the sub-dust case
  now expects `ErrFeeInputProducesDust`; added a boundary case where change ==
  `DustLimit` is still accepted (one output emitted).
- Added `TestCPFPChildSelectionAvoidsDustOutput`: with a poison near-dust UTXO
  next to a large one, selecting for `totalFee` reproduces the bad pick (now
  refused at build), while selecting for `totalFee + DustLimit` lands on the
  large input and yields a valid single-output child.

## Test plan

- [x] `go build ./txconfirm/`
- [x] `go test ./txconfirm/ -count=1`
- [x] `go vet ./txconfirm/`
- [x] `sg scan` — no new style warnings introduced

### CODE REVIEW

**Comment by Roasbeef** [comment]
@claude review this

File: txconfirm/broadcaster.go [hunk 1]
```diff
@@ -684,7 +693,14 @@ func (b *CPFPBroadcaster) broadcastWithCPFP(ctx context.Context, height int32,
...
 
-	feeInput, err := b.selectFeeInput(ctx, txid, totalFee)
+	// Select a fee input that covers the package fee *plus* a spendable
+	// (non-dust) change output. Selecting for totalFee alone can pick a
+	// UTXO whose post-fee remainder is below the dust limit, which would
+	// force BuildCPFPChild to drop its only output and produce a
+	// zero-output child that FinalizePsbt rejects. Requiring the dust
+	// buffer up front skips near-dust UTXOs and lands on one that always
+	// yields a valid change output.
+	feeInput, err := b.selectFeeInput(ctx, txid, totalFee+DustLimit)
```
[the remark this answers is not in the document]
Comment (Roasbeef) [reply txconfirm/broadcaster.go:703]: Addressed in the latest push (folded into the `txconfirm:` commit).

After the precise per-input vsize recalculation grows `totalFee`, the
broadcaster now reselects against the updated `totalFee + DustLimit`
threshold before calling `BuildCPFPChild`:

- If the originally selected input still covers the larger fee, the
  reservation cache returns it unchanged — common path is untouched.
- If it no longer covers it, a larger fee input is chosen (and reserved)
  so the child still carries a spendable change output.
- If no input satisfies the precise threshold, we fall through to
  `fallbackDirectBroadcast` with the original outpoint.

This closes the mixed-type-wallet retry loop where a P2PKH input would
trip `ErrFeeInputProducesDust` and get re-picked every retry. Regression
test `TestCPFPReselectsAfterPreciseFeeGrowth` pins the behavior: a small
P2PKH UTXO that clears the initial threshold but not the grown one is
selected first, then abandoned in favor of the larger UTXO.

**Comment by Roasbeef** [comment]
@claude review this
